src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -3752,10 +3752,14 @@ private void recordRelocation(void* location, void* locationRW, void* target, us
                     break;
             }
 
-            relocDelta += addlDelta;
-
             TargetArchitecture targetArchitecture = _compilation.TypeSystemContext.Target.Architecture;
             RelocType relocType = GetRelocType(targetArchitecture, fRelocType);
+
+            if (relocType == RelocType.IMAGE_REL_BASED_REL32)
```

Comment:
Would it make more sense for the JIT to pass `addlDelta` as zero? 

One can imagine cases where `addlDelta` can be useful for other relocation types as well. They may not exist today, but they can show up in future.

Reply:
Maybe? That's what https://github.com/dotnet/runtime/pull/79467 does. I don't have a strong opinion either way.

I was interested to learn when I investigated that ngen ignored addlDelta except for REL32, and I'm generally wary of ways in which the crossgen2 implementation of the JIT/EE interface diverges from the ngen implementation.

Reply:
I think #79467 is a better fix.

I would not use ngen or .NET Framework as a gold standard for JIT/EE interface implementation. The JIT/EE interface contract in .NET Framework had a lot of things in it that did not make sense and it was a lot of friction to fix them.